src/main/java/se/liu/ida/hefquin/engine/query/impl/QueryPatternUtils.java

```diff
@@ -179,10 +208,56 @@ public static int getNumberOfVarOccurrences( final OpBGP bgp ) {
 		return n;
 	}
 
+	public static int getNumberOfVarOccurrences( final OpJoin join ) {
+		final int numLeft = getNumberOfVarOccurrences( join.getLeft() );
+		final int numRight = getNumberOfVarOccurrences( join.getRight() );
+
+		return numLeft+numRight;
+	}
+
+	public static int getNumberOfVarOccurrences( final Op op ) {
+		if ( op instanceof OpBGP ) {
+			return getNumberOfVarOccurrences( (OpBGP) op);
+		}
+		else if ( op instanceof OpJoin ) {
+			return getNumberOfVarOccurrences( (OpJoin) op );
+		}
+		else if ( op instanceof OpFilter )  {
+			// TODO
+			return 0;
```

Comment:
What is the "TODO" supposed to mean? Does it mean that this case still needs to be implemented? If so, please throw an UnsupportedOperationException instead of returning a number.

Reply:
I marked "TODO" here, to indicate that this number is still needed to be determined.

Reason: I was unsure whether we need to take into account the number of elements (variables and terms) in a FILTER expression in the cost model.
Returning 0 means that the variables and RDF terms in a filter expression are not considered in the total number of shipping data.

Reply:
I see. The question then is, what would you say? More precisely, from the perspective of the current cost model, does it make sense to include the number of elements (variables and terms) of a FILTER in the count for the corresponding cost dimensions?